src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -34,20 +42,26 @@ public void should_return_a_404_if_task_does_not_exist() {
             .body("error", equalTo(HttpStatus.NOT_FOUND.getReasonPhrase()))
             .body("status", equalTo(HttpStatus.NOT_FOUND.value()))
             .body("message", equalTo(String.format(
-                "There was a problem updating the task with id: %s. The task could not be found.",
+                LOG_MSG_THERE_WAS_A_PROBLEM_FETCHING_THE_VARIABLES_FOR_TASK,
                 nonExistentTaskId
             )));
     }
 
+    private String getAssigneeId(Header caseworkerBAuthorizationOnly) {
+        return authorizationHeadersProvider.getUserInfo(caseworkerBAuthorizationOnly.getValue()).getUid();
+    }
+
     @Test
     public void should_return_a_204_when_assigning_a_task_by_id() {
-
-
         Map<String, String> task = common.setupTaskAndRetrieveIds();
 
+        String assigneeId = getAssigneeId(authorizationHeadersProvider.getCaseworkerBAuthorizationOnly());
         Response result = restApiActions.post(
-            "task/{task-id}/assign",
+            ENDPOINT_BEING_TESTED,
             task.get("taskId"),
+            new AssigneeRequest(assigneeId),
+            APPLICATION_JSON_VALUE,
+            APPLICATION_JSON_VALUE,
             authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
         );
```

Comment:
should_return_a_204_when_assigning_a_task_by_id()  appears to cover AC1. 

We also need to cover AC2. i.e
- User has -an Organisational Role of "tribunal-caseworker" or "senior-tribunal-caseworker" & permission of "MANAGE"
- Assignee has  a Case Role of "tribunal-caseworker" &  the OWN or EXECUTE permission for this task

Are we still dependent on full ccd integration before we can test CASE ROLE? 

Alex mentioned this for the other endpoint access management tickets so i created a ticket (https://tools.hmcts.net/jira/browse/RWA-225) to implement functional tests for CASE ROLE AC's for all end point tickets where its not covered,  when full ccd integration its done.

I have linked RWA-162 to it.